Userland/Games/Chess/Engine.cpp

```diff
@@ -43,9 +43,15 @@ void Engine::connect_to_engine_service()
     posix_spawn_file_actions_adddup2(&file_actions, wpipefds[0], STDIN_FILENO);
     posix_spawn_file_actions_adddup2(&file_actions, rpipefds[1], STDOUT_FILENO);
 
-    char const* argv[] = { m_command.characters(), nullptr };
+    auto command_length = m_command.code_points().length();
+    auto command_name = new char[command_length + 1];
```

Comment:
We seem to be leaking this,
can we safely `delete[]` this after `posix_spawnp`?

(We also have `Core::System::posix_spawnp` but that does not help much in this case)

Reply:
Freeing it after creating the new process (`posix_spawnp`) does seem to work.